tests/lib/mountinfo-tool

```diff
@@ -0,0 +1,379 @@
+#!/usr/bin/env python3
+from abc import ABC, abstractmethod
+from argparse import ArgumentParser, FileType, RawTextHelpFormatter
+from typing import Dict, List, Tuple, Any, Type, Callable, Match
+import re
+
+
+class Device(int):
+    """
+    Device is a device number with major and minor components.
+
+    Note that this class does not attempt to mimic peculiar
+    encoding used by the Linux kernel.
+    """
+
+    @classmethod
+    def pack(cls, major: int, minor: int) -> 'Device':
+        return cls((major << 16) | (minor & (1 << 16) - 1))
+
+    def __str__(self) -> str:
+        return '{}:{}'.format(self.major, self.minor)
+
+    def __repr__(self) -> str:
+        return 'Device.pack({}, {})'.format(self.major, self.minor)
+
+    @property
+    def major(self) -> int:
+        """major is the higher 16 bits of the device number."""
+        return self >> 16
+
+    @property
+    def minor(self) -> int:
+        """minor is the lower 16 bits of the device number."""
+        return self & ((1 << 16) - 1)
```

Comment:
or simply `self & 0xffff` without being too fancy

Reply:
well... yeah though I don't think this is a blocker :)